Monika After Story/game/script-holidays.rpy

```diff
@@ -3262,3 +3265,416 @@ label greeting_d25p_returned_nydp:
     m "I'm really happy you took me."
     m 3hub "Happy New Year, [player]~"
     return
+
+########################################################### player_bday ########################################################################
+# [HOL040]
+
+# date of current year bday, accounting for leap years
+init 5 python:
+    global mas_player_bday_curr
+
+    if persistent._mas_player_bday is None:
+        mas_player_bday_curr = None
+
+    else:
+        mas_player_bday_curr = store.mas_utils.add_years(persistent._mas_player_bday,datetime.date.today().year-persistent._mas_player_bday.year)
+
+# so we know we are in player b_day mode
+default persistent._mas_player_bday_in_player_bday_mode = False
+# so we know if you ruined the surprise
+default persistent._mas_player_bday_opened_door = False
+# for various reason, no decorations
+default persistent._mas_player_bday_no_decor = False
+# number of bday dates
+default persistent._mas_player_bday_date = 0
+# so we know if returning home post bday it was a bday date
+default persistent._mas_player_bday_left_on_bday = False
+# affection gained on bday dates
+default persistent._mas_player_bday_date_aff_gain = 0
+
+init -10 python:
+    def mas_isplayer_bday(_date=None):
+        if _date is None:
+            _date = datetime.date.today()
+
+        if persistent._mas_player_bday is None:
+            return False
+        else:
+            return _date == store.mas_utils.add_years(persistent._mas_player_bday,_date.year-persistent._mas_player_bday.year)
+
+init -810 python:
+    # MASHistorySaver for player_bday
+    store.mas_history.addMHS(MASHistorySaver(
+        "player_bday",
+        datetime.datetime(2020, 1, 6),
```

Comment:
a couple of things:

With the trigger in early jan, you would want to have `use_year_before` set to True here to get the correct year. However that fails if the user's bday is between jan1 to 6. So really this should trigger on jan 1, with the `use_year_before` set to True.